### DISCUSSION TITLE
Save to directly rsync dbase?

### DISCUSSION THREAD

**Discussion by shaderecker** [shaderecker/ansible-pihole#2] [General]
@deHakkelaar brought this up in [Discourse](https://discourse.pi-hole.net/t/high-availability-ha-for-pi-hole-running-two-pi-holes/3138/75)

I noticed you do a one on one `rsync` of the gravity dbase file:

https://github.com/shaderecker/ansible-pihole/blob/d6b2f538b221536c0f03b6b73aa60d5ce79f5dcc/roles/sync/templates/pihole_sync.j2#L5-L12

Wouldn't that potentially also `rsync` the dbase file that might still be in transition performing some `UPDATE` query or something else ?
Isnt it safer to dump the dbase instead on the "primary" and copy the dump over to the "secondaries" eg:

`sqlite3 /etc/pihole/gravity.db ".backup main gravity.db.dump"`

And restore/import on the secondary hosts if checksum check (md5 or whatever) has changed with:

`sudo -u pihole sqlite3 /etc/pihole/gravity.db ".restore main gravity.db.dump"`

~~Would need to stop pihole-FTL to prevent tables being locked while importing.~~

**Comment by deHakkelaar** [comment]
Both rsync and the gravity pull could run at the same time which could result in tables still being in transition while the file already is being rsynced.
Am not sure if its safe to copy a dbase file with rsync this way.
I believe the recommended way is to dump and import again at the other end when concerning dbases.
Or some other way to sync via SQL.

**Reply by shaderecker** [reply]
Thanks again for the suggestions. I actually didn't think about that case yet.
I will have a look and adapt my script.

**Comment by shaderecker** [comment]
Finally came around to implement this in #7 😁
